hudi-utilities/src/main/java/org/apache/hudi/utilities/schema/RowBasedSchemaProvider.java

```diff
@@ -44,7 +45,12 @@ public RowBasedSchemaProvider(StructType rowStruct) {
 
   @Override
   public Schema getSourceSchema() {
-    return AvroConversionUtils.convertStructTypeToAvroSchema(rowStruct, HOODIE_RECORD_STRUCT_NAME,
-        HOODIE_RECORD_NAMESPACE);
+    try {
+      return AvroConversionUtils.convertStructTypeToAvroSchema(rowStruct, HOODIE_RECORD_STRUCT_NAME,
+          HOODIE_RECORD_NAMESPACE);
+    } catch (Exception e) {
+      throw new HoodieSchemaException("Failed to convert struct type to avro schema", e);
```

Comment:
Should we move this throw into the AvroConversionUtils? I see other methods in there throw the exception themselves.

Reply:
Good catch. I actually do throw it in there.